File: Semester 4/Class13_Computer_Contest_Level_3_Notes_and_Homework_Dec_03/lvl3_hw_sol/boring_business.cpp
#include <bits/stdc++.h>
using namespace std;
bool vis[200][401]; int curR = 0, curC = 0, len; char dir;
bool drill(char direction, int length){
    int dr = 0, dc = 0; bool safe = true;
    if(direction == 'u') dr = 1;
    if(direction == 'd') dr = -1;
    if(direction == 'l') dc = -1;
    if(direction == 'r') dc = 1;
    for(int i=0; i<length; i++) {
        curR += dr; curC += dc;
        if(vis[curR+200][curC+200]) safe = false;
        vis[curR+200][curC+200] = true;
    }
    return safe;
}
int main(){
    drill('d', 3); drill('r', 3); drill('d', 2); drill('r', 2); drill('u', 2); drill('r', 2);
    drill('d', 4); drill('l', 8); drill('u', 2);
    while(cin >> dir >> len && dir != 'q'){
        if(drill(dir, len)) cout << curC << " " << curR << " safe" << endl;
        else {
            cout << curC << " " << curR << " DANGER" << endl;
            break;
        }
    }
}

File: Semester 4/Class13_Computer_Contest_Level_3_Notes_and_Homework_Dec_03/lvl3_hw_sol/boring_business.java
import java.util.*;
import java.io.*;
public class homework {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringTokenizer st;
	static int r = 0, c = 0;
	static int[][] arr = new int[205][405];
	static boolean safe = true;
	public static void main(String[] args) throws IOException{
        drill('d', 3);  drill('r', 3); drill('d', 2); drill('r', 2); drill('u', 2); drill('r', 2); 
        drill('d', 4); drill('l', 8); drill('u', 2);
        while(safe){
            char direction = readCharacter();
            int stp = readInt();
            if(direction == 'q') break;
            else{
                 drill(direction, stp);
                 System.out.print(c + " " + r);
                 if(safe){
                     System.out.println(" safe");
                 }else System.out.println(" DANGER");
            }
	    }
	}
	static void drill(char dir, int step){
	    int dr = 0, dc = 0;
	    if(dir == 'd') dr=-1;
	    if(dir == 'u') dr=1;
	    if(dir == 'l') dc=-1;
	    if(dir == 'r') dc=1;
	    for(int i = 0; i < step; i++){
	        r += dr;
	        c += dc;
	        if(arr[r+200][c+200] != 0){
	            safe = false;
	        }else{
	            arr[r+200][c+200]=1;
	        }
	    }
	}
	static String next () throws IOException {
		while (st == null || !st.hasMoreTokens())
			st = new StringTokenizer(br.readLine().trim());
		return st.nextToken();
	}
	static long readLong () throws IOException {
		return Long.parseLong(next());
	}
	static int readInt () throws IOException {
		return Integer.parseInt(next());
	}
	static double readDouble () throws IOException {
		return Double.parseDouble(next());
	}
	static char readCharacter () throws IOException {
		return next().charAt(0);
	}
	static String readLine () throws IOException {
		return br.readLine().trim();
	}
}